packages/node_modules/@webex/internal-plugin-device/test/unit/spec/device.js

```diff
@@ -178,5 +179,164 @@ describe('plugin-device', () => {
         );
       });
     });
+
+    describe('#refresh()', () => {
+      let requestSpy;
+
+      const setup = () => {
+        sinon.stub(device, 'canRegister').callsFake(() => Promise.resolve());
+        sinon.stub(device, 'processRegistrationSuccess').callsFake(() => {});
+        requestSpy = sinon.spy(device, 'request');
+        device.config.defaults = {};
+        device.set('registered', true);
+      };
+
+      it('If-None-Match header is added if etag is set', async () => {
+        setup();
+
+        device.set('etag', 'etag-value');
+
+        const result = device.refresh();
+
+        await result;
+
+        assert.deepEqual(requestSpy.args[0][0].headers, {
+          'If-None-Match': 'etag-value'
+        });
+      });
+
+      it('If-None-Match header is not added if etag is not set', async () => {
+        setup();
+
+        const result = device.refresh();
+
+        await result;
+
+        assert.deepEqual(requestSpy.args[0][0].headers, {});
+      });
+    });
+
+    describe('#processRegistrationSuccess()', () => {
+      let clonedDTO;
+
+      const setup = () => {
+        clonedDTO = cloneDeep(dto);
+        clonedDTO.features = {
```

Comment:
nitpick: It's odd having all of these returned all of the time - would be nice to match the expected server side behaviour (i.e. only return user features if etag matches).